examples/simple_example.rs

```diff
@@ -0,0 +1,136 @@
+//! Centralized Telescope example with BLS signatures
+
+use alba::centralized_telescope::params::Params;
+use alba::centralized_telescope::proof::Proof;
+use alba::centralized_telescope::CentralizedTelescope;
+use blst::min_sig::{AggregatePublicKey, AggregateSignature, PublicKey, SecretKey, Signature};
+use blst::BLST_ERROR;
+use rand_chacha::ChaCha20Rng;
+use rand_core::{CryptoRng, RngCore, SeedableRng};
+use std::collections::HashMap;
+
+const DATA_LENGTH: usize = 48;
+pub(crate) type Element = [u8; DATA_LENGTH];
+
+struct AlbaSigner {
+    signing_key: SecretKey,
+    verification_key: PublicKey,
+}
+
+struct ThresholdSignature {
+    proof: Proof,
+    key_list: Vec<PublicKey>,
+}
+
+impl AlbaSigner {
+    fn new(rng: &mut (impl RngCore + CryptoRng)) -> Self {
+        let mut ikm = [0u8; 32];
+        rng.fill_bytes(&mut ikm);
+        let sk = SecretKey::key_gen(&ikm, &[])
+            .expect("Error occurs when the length of ikm < 32. This will not happen here.");
+        let pk: PublicKey = sk.sk_to_pk();
+        Self {
+            signing_key: sk,
+            verification_key: pk,
+        }
+    }
+
+    fn sign<const N: usize>(&self, msg: &[u8]) -> [u8; N] {
+        let mut signature_to_byte = [0u8; N];
+        signature_to_byte.copy_from_slice(&self.signing_key.sign(msg, &[], &[]).to_bytes());
+        signature_to_byte
+    }
+}
+
+impl ThresholdSignature {
+    fn aggregate<const N: usize>(
+        alba_signatures: &HashMap<Element, usize>,
+        params: &Params,
+        key_list: &HashMap<usize, PublicKey>,
+    ) -> Self {
+        let prover_set: Vec<Element> = alba_signatures.keys().copied().collect();
+        let alba = CentralizedTelescope::create(params);
+        let proof = alba.prove(&prover_set).unwrap();
+        let signatures = proof.element_sequence.clone();
+        let mut public_keys = Vec::with_capacity(signatures.len());
+
+        for sig in signatures {
+            public_keys.push(
+                *key_list
+                    .get(alba_signatures.get(sig.as_slice()).unwrap())
+                    .unwrap(),
+            );
+        }
+        Self {
+            proof,
+            key_list: public_keys,
+        }
+    }
+
+    /// Validates individual signatures in the threshold signature
+    fn validate_signatures(&self, msg: &[u8]) -> bool {
+        let mut signatures = Vec::with_capacity(self.proof.element_sequence.len());
+        for sig_bytes in &self.proof.element_sequence {
+            if let Ok(signature) = Signature::from_bytes(sig_bytes.as_slice()) {
+                signatures.push(signature);
+            } else {
+                println!("Error: Failed to parse signature from bytes.");
+                return false;
+            }
+        }
+        let signature_refs: Vec<&Signature> = signatures.iter().collect();
+        let aggregate_signature =
```

Comment:
Similarly
```
let Ok(signature) = AggregateSignature::aggregate(signature_refs.as_slice(), false) else { return false };
agg_sig.to_signature();
```

Reply:
In order to make this code work, we need to define another variable to hold the signature, which is created with `to_signature()`. Like following: 
```
let Ok(aggregate_signature) =
            AggregateSignature::aggregate(signature_refs.as_slice(), false)
        else {
            return false;
        };
let final_signature = aggregate_signature.to_signature();
```
Because the function `to_signature()` takes an `AggregateSignature` as input and returns a `Signature`.

Reply:
Sorry, I wrongly thought that to_signature was an in place function I understand now!
It just becomes a matter of coding style then which is not very important. I will resolve this comment now and leave you to choose which style you prefer.